cardano-testnet/src/Testnet/EpochStateProcessing.hs

```diff
@@ -75,3 +83,35 @@ maybeExtractGovernanceActionIndex sbe txid (AnyNewEpochState actualEra newEpochS
     compareWithTxId (Api.TxId ti1) Nothing (GovActionId (TxId ti2) (L.GovActionIx gai)) _
       | ti1 == L.extractHash ti2 = Just gai
     compareWithTxId _ x _ _ = x
+
+-- | Watch the epoch state view until the guard function returns @Just@ or the timeout is reached.
+-- | Wait for at least `minWait` epochs and at most `maxWait` epochs.
+-- | The function will return the result of the guard function if it is met, otherwise it will return @Nothing@.
+watchEpochStateView
+  :: forall m a. (MonadIO m, MonadTest m, MonadAssertion m)
+  => EpochStateView -- ^ The info to access the epoch state
+  -> (AnyNewEpochState -> m (Maybe a)) -- ^ The guard function (@Just@ if the condition is met, @Nothing@ otherwise)
+  -> Word64 -- ^ The minimum number of epochs to wait
+  -> Word64 -- ^ The maximum number of epochs to wait
+  -> m (Maybe a)
+watchEpochStateView epochStateView f minWait maxWait = do
+  AnyNewEpochState _ newEpochState <- getEpochState epochStateView
+  let (EpochNo currentEpoch) = L.nesEL newEpochState
+  go (EpochNo $ currentEpoch + minWait) (EpochNo $ currentEpoch + maxWait)
+    where
+      go :: EpochNo -> EpochNo -> m (Maybe a)
+      go (EpochNo startEpoch) (EpochNo timeout) = do
+        epochState@(AnyNewEpochState _ newEpochState') <- getEpochState epochStateView
+        let (EpochNo currentEpoch) = L.nesEL newEpochState'
+        if currentEpoch < startEpoch
+        then do H.threadDelay 100_000
```

Comment:
This is to test both changes expected to occur and changes expected to _not_ occur right?

https://github.com/IntersectMBO/cardano-node/pull/5815#discussion_r1592978314

Reply:
I will remove the `minWait` parameter as suggested by [this review](https://github.com/IntersectMBO/cardano-node/pull/5830#discussion_r1595196951), and, after that, it will just be a function that does a generic test about the value of epoch state. That should make it simpler and easier to understand.

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5830/commits/228b1a87d347ef7bd4237b48f87ff7532475725d